### ISSUE TITLE
Allow iterator and pytorch dataloader as input in any attribution algorithm

### ISSUE THREAD

**Issue by hal-314** [pytorch/captum#494]
## 🚀 Feature
Allow iterator and pytorch dataloader as input in any attribution algorithm

## Motivation

Currently, you can't easily estimate attributions for more samples that fit in your GPU or RAM. For example, in image classification, computing attributions for all images of certain label (dog, cat, person, etc) .  

## Pitch
Allow passing a custom iterators or a pytorch dataloaders instead of tensors as input in any attribution algorithm. So, users could compute attributions for an arbitrary number of samples. I include iterators as well as pytorch dataloaders because in certain cases, you need to perform extra steps / data transformations to feed the batch as captum algorithm input. 

For example, when a model has embeddings like BERT that are replaced by `InterpretableEmbeddingsBase`, you need to map token indexes from dataloader to embeddings values before passing them as input to algorithm attribute method. For instance, you want to analyze attribution for BERT model with Integrated Gradients.

## Alternatives

The current workaround is manually writing a loop that iterates over all your dataset / samples and concatenate attributions. It seems awkward since pytorch has dataloaders and captum could do that.

**Comment by vivekmig** [comment]
Hi @hal-314 , thanks for the feature request, and apologies for the long delay in addressing this feature request! @aobo-y has recently worked on adding this functionality, an initial version can be found here #1155
